examples/miscellaneous/plot_metadata_routing.py

```diff
@@ -373,56 +397,64 @@ def predict(self, X, **predict_params):
 
 # %%
 # - No metadata requested
-est = RouterConsumerClassifier(estimator=ExampleClassifier())
-print_routing(est)
+meta_est = RouterConsumerClassifier(estimator=ExampleClassifier())
+print_routing(meta_est)
 
 
 # %%
 # - ``sample_weight`` requested by underlying estimator
-est = RouterConsumerClassifier(
+meta_est = RouterConsumerClassifier(
     estimator=ExampleClassifier().set_fit_request(sample_weight=True)
 )
-print_routing(est)
+print_routing(meta_est)
 
 # %%
 # - ``sample_weight`` requested by meta-estimator
-est = RouterConsumerClassifier(estimator=ExampleClassifier()).set_fit_request(
+meta_est = RouterConsumerClassifier(estimator=ExampleClassifier()).set_fit_request(
     sample_weight=True
 )
-print_routing(est)
+print_routing(meta_est)
 
 # %%
 # Note the difference in the requested metadata representations above.
 #
 # - We can also alias the metadata to pass different values to them:
 
-est = RouterConsumerClassifier(
+meta_est = RouterConsumerClassifier(
     estimator=ExampleClassifier().set_fit_request(sample_weight="clf_sample_weight"),
 ).set_fit_request(sample_weight="meta_clf_sample_weight")
-print_routing(est)
+print_routing(meta_est)
 
 # %%
 # However, ``fit`` of the meta-estimator only needs the alias for the
 # sub-estimator, since it doesn't validate and route its own required metadata:
-est.fit(X, y, sample_weight=my_weights, clf_sample_weight=my_other_weights)
+meta_est.fit(
+    X, y, sample_weight=my_weights, clf_sample_weight=my_other_weights
+)  # <-- `meta_clf_sample_weight=my_weights`?
```

Comment:
I find it very confusing why this would throw an error:

`meta_est.fit(X, y, meta_clf_sample_weight=my_weights, clf_sample_weight=my_other_weights) `

From the wording ("only needs") it doesn't seem so strict, and I also don't understand the reasoning.